build/Dockerfile

```diff
@@ -35,19 +35,22 @@ WORKDIR /build/windows_exporter/
 COPY windows_exporter/ .
 RUN GOOS=windows /build/promu/promu build -v
 
+# Build CNI plugins
+WORKDIR /build/containernetworking-plugins/
+COPY containernetworking-plugins/ .
+ENV CGO_ENABLED=0
+RUN ./build_windows.sh
+
 # Build Kubernetes node binaries
 WORKDIR /build/kubernetes/
 COPY kubernetes/ .
+RUN rm /build/kubernetes/.git
+# Copy the git metadata required for setting the kubelet and kube-proxy version. Sometimes misses container cache.
+COPY .git/modules/kubernetes /build/kubernetes/.git
```

Comment:
I suggest restructuring the file layout to be `/build/windows-machine-config-operator/$sub-module`. All you would copy into the `build/windows-machine-config-operator` would be the full main `.git`.

Reply:
copying .git will be the first step in that case. The main culprit resetting the cache in our case is the .git folder. What i am doing here is, just copying the metadata needed for kubernetes repo. And not for other sub-modules.